federation/src/main/scala/caliban/federation/FederationHelpers.scala

```diff
@@ -39,26 +40,24 @@ private[federation] object FederationHelpers {
 
   val anyArgBuilder: ArgBuilder[_Any] = {
     case v @ InputValue.ObjectValue(fields) =>
-      fields
-        .get("__typename")
-        .collect { case StringValue(__typename) =>
-          _Any(__typename, v)
-        }
-        .toRight(ExecutionError("_Any must contain a __typename value"))
+      fields.getOrElseNull("__typename") match {
+        case StringValue(__typename) => Right(_Any(__typename, v))
+        case _                       => Left(ExecutionError("_Any must contain a __typename value"))
+      }
     case other                              => Left(ExecutionError(s"Can't build a _Any from input $other"))
   }
 
   case class RepresentationsArgs(representations: List[_Any])
 
   implicit val representationsArgBuilder: ArgBuilder[RepresentationsArgs] = {
     case InputValue.ObjectValue(fields) =>
-      fields.get("representations").toRight(ExecutionError("_Any must contain a __typename value")).flatMap {
+      fields.getOrElseNull("representations") match {
         case InputValue.ListValue(values) =>
           traverseEither(values.map(anyArgBuilder.build)).map(RepresentationsArgs.apply)
+        case null                         => Left(ExecutionError("_Any must contain a __typename value"))
```

Comment:
Error message seems off

Reply:
Changed it, not sure if it's the best but kept it aligned with `_Any`